src/components/AppProvider/AppProvider.tsx

```diff
@@ -0,0 +1,62 @@
+import React, { useState, useMemo } from 'react';
+import { Movie } from '../../types/Movie';
+
+type Context = {
+  movies: Movie[],
+  setMovies: (newMovies: Movie[]) => void,
+  query: string,
+  setQuery: (newQuery: string) => void,
+  movie: Movie | null,
+  setMovie: (newMovie: Movie | null) => void,
+  isMovieExist: boolean,
+  setIsMovieExist: (newCheck: boolean) => void,
+  isSearch: boolean,
+  setIsSearch: (newStatus: boolean) => void,
+};
+
+export const AppContext = React.createContext<Context>({
+  movies: [],
+  setMovies: () => {},
+  query: '',
+  setQuery: () => {},
+  movie: null,
+  setMovie: () => {},
+  isMovieExist: false,
+  setIsMovieExist: () => {},
+  isSearch: false,
+  setIsSearch: () => {},
+});
+
+export const AppProvider: React.FC = ({ children }) => {
+  const [movies, setMovies] = useState<Movie[]>([]);
+  const [query, setQuery] = useState('');
+  const [movie, setMovie] = useState<Movie | null>(null);
+  const [isMovieExist, setIsMovieExist] = useState(false);
+  const [isSearch, setIsSearch] = useState(false);
+
+  const contextValue = useMemo(() => ({
```

Comment:
Using useMemo without good reason is bad practice, overall this is strange use case to memoize function that did not calculate anything and just returns object of const values